prover/src/provers/agg_prover.rs

```diff
@@ -87,6 +95,49 @@ impl Prover<AggContext> for AggProver {
         let json_string = serde_json::to_string(&updated_agg_public_values)?;
         let _ = file::new(&agg_pub_value_path).write(json_string.as_bytes())?;
 
+        if ctx.is_final {
+            let (block_proof, _block_public_values) =
+                all_circuits.prove_block(None, &agg_proof, updated_agg_public_values)?;
+
+            log::info!(
+                "proof size: {:?}",
+                serde_json::to_string(&block_proof.proof).unwrap().len()
+            );
+            let _result = all_circuits.verify_block(&block_proof);
+
+            let path = output_dir.to_string();
+            let builder = WrapperBuilder::<DefaultParameters, 2>::new();
+            let mut circuit = builder.build();
+            circuit.set_data(all_circuits.block.circuit);
+            let wrapped_circuit =
+                WrappedCircuit::<InnerParameters, OuterParameters, D>::build(circuit);
+            println!("build finish");
+
+            let wrapped_proof = wrapped_circuit.prove(&block_proof).unwrap();
+            // save wrapper_proof
+            file::new(&path).create_dir_all()?;
+            let common_data_file = if path.ends_with('/') {
+                format!("{}common_circuit_data.json", path)
+            } else {
+                format!("{}/common_circuit_data.json", path)
+            };
+            let _ = file::new(&common_data_file)
+                .write(&serde_json::to_vec(&wrapped_proof.common_data)?)?;
+            let verify_data_file = if path.ends_with('/') {
+                format!("{}verifier_only_circuit_data.json", path)
+            } else {
+                format!("{}/verifier_only_circuit_data.json", path)
+            };
+            let _ = file::new(&verify_data_file)
```

Comment:
we don't have to receive the result, just remove `let _ = `

Reply:
If removed, an error will be reported
'error: written amount is not handled'